src/metpy/plots/declarative.py

```diff
@@ -1262,3 +1265,244 @@ def _build(self):
+        return self._obsdata
+
+    @property
+    def plotdata(self):
+        """Return the data for plotting.
+
+        The data arrays, x coordinates, and y coordinates.
+
+        """
+        plot_data = {}
+        for dim_name in list(self.obsdata):
+            if dim_name.find('lat') != -1:
+                lat = self.obsdata[dim_name]
+            elif dim_name.find('lon') != -1:
+                lon = self.obsdata[dim_name]
+            else:
+                plot_data[dim_name] = self.obsdata[dim_name]
+        return lon.values, lat.values, plot_data
+
+    def draw(self):
+        """Draw the plot."""
+        if self._need_redraw:
+            if getattr(self, 'handle', None) is None:
+                self._build()
+            self._need_redraw = False
+
+    @observe('colors', 'symbol_mapper', 'locations', 'reduce_points')
+    def _set_need_rebuild(self, _):
+        """Handle changes to attributes that need to regenerate everything."""
+        # Because matplotlib doesn't let you just change these properties, we need
+        # to trigger a clear and re-call of contour()
+        self.clear()
+
+    def _build(self):
+        """Build the plot by calling needed plotting methods as necessary."""
+        lon, lat, data = self.plotdata
+
+        # Use the cartopy map projection to transform station locations to the map and
+        # then refine the number of stations plotted by setting a 300km radius
+        if self.parent._proj_obj == ccrs.PlateCarree():
+            scale = 1.
+        else:
+            scale = 100000.
+        point_locs = self.parent._proj_obj.transform_points(ccrs.PlateCarree(), lon, lat)
+        subset = reduce_point_density(point_locs, self.reduce_points * scale)
+
+        stationplot = StationPlot(self.parent.ax, lon[subset], lat[subset], clip_on=True,
+                                  transform=ccrs.PlateCarree(), fontsize=10)
+
+        for i, ob_type in enumerate(self.fields):
+            if len(self.locations) > 1:
+                location = self.locations[i]
+            else:
+                location = self.locations[0]
+            if len(self.colors) > 1:
+                color = self.colors[i]
+            else:
+                color = self.colors[0]
+            if self.formats[i] is not None:
+                mapper = getattr(wx_symbols, str(self.formats[i]), None)
+                if mapper is not None:
+                    self.handle = stationplot.plot_symbol(location, data[ob_type][subset],
+                                                          mapper, color=color)
+                else:
+                    self.handle = stationplot.plot_parameter(location, data[ob_type][subset],
+                                                             color=color,
+                                                             formatter=self.formats[i])
+            else:
+                self.handle = stationplot.plot_parameter(location, data[ob_type][subset],
```

Comment:
I think `self.handle` for this one should just be set to `stationplot` rather than the parts that are individually created for each location.

Reply:
The difficulty I'm running into with this one is then how can be clear the plot for modifications in a more dynamic future scenario. It's built similar to the other declarative plotting code. I tried a couple of different iterations to try to remove the `self.handle` piece, but couldn't find a working solution. If we can figure out a way to clear the pieces, then I'm all game.

Reply:
Well the problem currently is that you keep resetting `self.handle` as you loop over the fields, so the code as is only has the last thing you plot. Also, Codacy pointed out that `plot_barbs` doesn't actually return anything, so if you add a vector you're setting self.handle to `None`.

If the tests still pass, I'd say do `self.handle = stationplot`, which *should* have what we need, but until we try it in a dynamic setting, who knows.

Reply:
Okay, I think I got it. It simplified a few things. Had to update the test images as they really weren't right in the first place for two of the new ones I had added. Now they are correct.